src/metabase/query_analysis.clj

```diff
@@ -53,56 +62,60 @@
   [& body]
   `(with-execution* ::disabled ~@body))
 
-(defn- execution []
+(defn- execution
+  "The execution strategy for analysis, which can be overridden in dev and tests. In production, it is always async."
+  []
   (case config/run-mode
     :prod ::queued
     :dev  *analyze-execution-in-dev?*
     :test *analyze-execution-in-test?*))
 
-
 (defn enabled-type?
   "Is analysis of the given query type enabled?"
   [query-type]
-  (case query-type
-    :native     (public-settings/sql-parsing-enabled)
-    :query      true
-    :mbql/query true
-    false))
+  (and (public-settings/query-analysis-enabled)
+       (case query-type
+         :native (public-settings/sql-parsing-enabled)
+         :query true
+         :mbql/query true
+         false)))
 
 (defn- query-field-ids
   "Find out ids of all fields used in a query. Conforms to the same protocol as [[query-analyzer/field-ids-for-sql]],
   so returns `{:explicit #{...int ids}}` map.
 
-  Does not track wildcards for queries rendered as tables afterwards."
-  [query]
-  (let [query-type (lib/normalized-query-type query)]
-    (when (enabled-type? query-type)
-      (case query-type
-        :native     (try
-                      (nqa/field-ids-for-native query)
-                      (catch Exception e
-                        (log/error e "Error parsing SQL" query)))
-        :query      {:explicit (mbql.u/referenced-field-ids query)}
-        :mbql/query {:explicit (lib.util/referenced-field-ids query)}))))
+  Does not track wildcards for queries rendered as tables afterward."
```

Comment:
TIL afterward is American English!

Reply:
I'm running a grammar plugin to slowly learn this strange dialect 😄